lib/common/memory.c

```diff
@@ -347,7 +348,8 @@ void h2o_vector__expand(h2o_mem_pool_t *pool, h2o_vector_t *vector, size_t eleme
     while (vector->capacity < new_capacity)
         vector->capacity *= 2;
     if (pool != NULL) {
-        new_entries = h2o_mem_alloc_pool(pool, element_size * vector->capacity);
+        new_entries =
+            h2o_mem_alloc_pool_aligned(pool, 2 * sizeof(void *), H2O_ALIGN(element_size * vector->capacity, 2 * sizeof(void *)));
```

Comment:
How about passing the alignment requirement as an argument to `h2o_vector__expand`? Call to the function is wrapped via `h2o_vector_reserve` in which you can get the name of the typed element.

Reply:
fixed. please check commit 4dd533.